tests/integration/run_integration_test.sh

```
@@ -428,6 +428,26 @@ else
     pass "Write correctly rejected (read-only filesystem)"
 fi
 
+# 4h. Test: Duplicated URL deduplication
+log_info "Test group: Duplicated URL deduplication"
+
+# Count the occurrences of 'simple.txt' in the mounted directory listing
+simple_count=$(ls -1 "${MOUNT_DIR}" | grep -Fx "simple.txt" | wc -l)
```

Shellcheck (0.11.0)

[warning] Don't use ls | grep. Use a glob or a for loop with a condition to allow non-alphanumeric filenames.

(SC2010)